src/main/java/skyler/Storage.java

```diff
@@ -0,0 +1,232 @@
+            }
+            currPath += "/";
+        }
+
+        String filePath = new File(currPath).getAbsolutePath();
+        File directory = new File(filePath);
+
+        String[] flist = directory.list();
+        int flag = 0;
+        // directory is empty
+        if (flist == null) {
+            filePath = new File(currPath + file).getAbsolutePath();
+            directory = new File(filePath);
+            directory.createNewFile();
+            throw new SkylerException();
+        }
+
+        for (int i = 0; i < flist.length; i++) {
+            String filename = flist[i];
+            if (filename.equals(file)) {
+                filePath = new File(currPath + file).getAbsolutePath();
+                directory = new File(filePath);
+                flag = 1;
+            }
+        }
+
+        // directory is not empty but specified file not found
+        if (flag == 0) {
+            filePath = new File(currPath + file).getAbsolutePath();
+            directory = new File(filePath);
+            directory.createNewFile();
+            throw new SkylerException();
+        }
+
+        Scanner sc = new Scanner(directory);
+        while (sc.hasNextLine()) {
+            String taskStr = sc.nextLine();
+            String taskDescription = taskStr.substring(8);
+
+            char type = taskStr.charAt(0);
+            int taskStatus = Character.getNumericValue(taskStr.charAt(4));
+
+            switch(type) {
+            case 'T':
+                Todo todo = new Todo(taskDescription);
+                if (taskStatus == 1) {
+                    todo.markAsDone();
+                }
+                tasks.add(todo);
+                break;
+            case 'D':
+                Pattern p = Pattern.compile(" [|] ");
+                String[] descriptors = p.split(taskDescription);
+
+                LocalDateTime dt = processDateTime(descriptors[1]);
+                Deadline deadline = new Deadline(descriptors[0], dt);
+                if (taskStatus == 1) {
+                    deadline.markAsDone();
+                }
+                tasks.add(deadline);
+                break;
+            case 'E':
+                Pattern p1 = Pattern.compile(" [|] ");
+                String[] descriptors1 = p1.split(taskDescription);
+
+                LocalDateTime dt1 = processDateTime(descriptors1[1]);
+                Event event = new Event(descriptors1[0], dt1);
+                if (taskStatus == 1) {
+                    event.markAsDone();
+                }
+                tasks.add(event);
+                break;
+            default:
+            }
+        }
+
+        return tasks;
+    }
+
+    /**
+     * Saves tasks to file
+     *
+     * @param task ArrayList of tasks to save to file.
+     * @throws IOException If IO operation fails.
+     */
+    public void saveTask(ArrayList<Task> task) throws IOException {
+        String filepath = new File(filePathChosen).getAbsolutePath();
```

Comment:
Just a nitpick but perhaps this variable could be renamed to `filePath` to keep it consistent with other `filePath` variables in the code